odoo/addons/base/models/ir_ui_view.py

```diff
@@ -360,13 +360,14 @@ def reset_arch(self, mode='soft'):
         """
         for view in self:
             arch = False
+            # Don't save current arch in previous since we reset, this arch is probably broken
+            view_with_context = view.with_context(no_save_prev=True, lang=None)
             if mode == 'soft':
                 arch = view.arch_prev
+                view_with_context.write({'arch_db': arch})
             elif mode == 'hard' and view.arch_fs:
                 arch = view.with_context(read_arch_from_file=True, lang=None).arch
-            if arch:
```

Comment:
I don't think it's a good idea to remove the `if arch:` check here. If a view has no previous `arch_db` (`arch_prev`), this allows setting `False` on "soft" reset, no ?